.github/workflows/next-e2e-test.yml

```
@@ -1,85 +1,86 @@
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: .github/paths-filter.yml
+
   e2e:
-    name: Next (E2E)
+    needs: filter
+    if: (matrix.app == 'next' && needs.filter.outputs.next == 'true') || (matrix.app == 'fastify' && needs.filter.outputs.api == 'true')
```

actionlint (1.7.11)

[error] context "matrix" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)